### PR TITLE
fix: macos targetting with native error handling

### LINKED ISSUE #3070: [macOS / NativeAOT] Build error since 4.0.0-beta2
### Package

Sentry

### .NET Flavor

.NET

### .NET Version

8.0.100

### OS

macOS

### SDK Version

4.0.0-beta3 and newer

### Self-Hosted Sentry Version

_No response_

### Steps to Reproduce

1. Create new macOS app: `dotnet new macos`
2. Add `<PublishAot>true</PublishAot>` to .csproj
3. Add `<PackageReference Include="Sentry" Version="4.0.0-beta.8" /> `to .csproj
4. Add `Sentry.SentrySdk.Init(o => o.Dsn = "dsn");` somewhere in code
5. `dotnet publish`

### Expected Result

App builds successfully

### Actual Result

Build error:
```
/usr/local/share/dotnet/packs/Microsoft.macOS.Sdk/14.2.8004/targets/Xamarin.Shared.Sdk.targets(1559,3): error : clang++ exited with code 1: [/Users/user/Desktop/Sentry_test/Sentry_test.csproj]
/usr/local/share/dotnet/packs/Microsoft.macOS.Sdk/14.2.8004/targets/Xamarin.Shared.Sdk.targets(1559,3): error : Undefined symbols for architecture x86_64: [/Users/user/Desktop/Sentry_test/Sentry_test.csproj]
/usr/local/share/dotnet/packs/Microsoft.macOS.Sdk/14.2.8004/targets/Xamarin.Shared.Sdk.targets(1559,3): error :   "_curl_easy_cleanup", referenced from: [/Users/user/Desktop/Sentry_test/Sentry_test.csproj]
/usr/local/share/dotnet/packs/Microsoft.macOS.Sdk/14.2.8004/targets/Xamarin.Shared.Sdk.targets(1559,3): error :       _sentry__curl_bgworker_state_free in libsentry-native.a(sentry_transport_curl.c.o) [/Users/user/Desktop/Sentry_test/Sentry_test.csproj]
/usr/local/share/dotnet/packs/Microsoft.macOS.Sdk/14.2.8004/targets/Xamarin.Shared.Sdk.targets(1559,3): error :   "_curl_easy_getinfo", referenced from: [/Users/user/Desktop/Sentry_test/Sentry_test.csproj]
/usr/local/share/dotnet/packs/Microsoft.macOS.Sdk/14.2.8004/targets/Xamarin.Shared.Sdk.targets(1559,3): error :       _sentry__curl_send_task in libsentry-native.a(sentry_transport_curl.c.o) [/Users/user/Desktop/Sentry_test/Sentry_test.csproj]
/usr/local/share/dotnet/packs/Microsoft.macOS.Sdk/14.2.8004/targets/Xamarin.Shared.Sdk.targets(1559,3): error :   "_curl_easy_init", referenced from: [/Users/user/Desktop/Sentry_test/Sentry_test.csproj]
/usr/local/share/dotnet/packs/Microsoft.macOS.Sdk/14.2.8004/targets/Xamarin.Shared.Sdk.targets(1559,3): error :       _sentry__curl_transport_start in libsentry-native.a(sentry_transport_curl.c.o) [/Users/user/Desktop/Sentry_test/Sentry_test.csproj]
/usr/local/share/dotnet/packs/Microsoft.macOS.Sdk/14.2.8004/targets/Xamarin.Shared.Sdk.targets(1559,3): error :   "_curl_easy_perform", referenced from: [/Users/user/Desktop/Sentry_test/Sentry_test.csproj]
/usr/local/share/dotnet/packs/Microsoft.macOS.Sdk/14.2.8004/targets/Xamarin.Shared.Sdk.targets(1559,3): error :       _sentry__curl_send_task in libsentry-native.a(sentry_transport_curl.c.o) [/Users/user/Desktop/Sentry_test/Sentry_test.csproj]
/usr/local/share/dotnet/packs/Microsoft.macOS.Sdk/14.2.8004/targets/Xamarin.Shared.Sdk.targets(1559,3): error :   "_curl_easy_reset", referenced from: [/Users/user/Desktop/Sentry_test/Sentry_test.csproj]
/usr/local/share/dotnet/packs/Microsoft.macOS.Sdk/14.2.8004/targets/Xamarin.Shared.Sdk.targets(1559,3): error :       _sentry__curl_send_task in libsentry-native.a(sentry_transport_curl.c.o) [/Users/user/Desktop/Sentry_test/Sentry_test.csproj]
/usr/local/share/dotnet/packs/Microsoft.macOS.Sdk/14.2.8004/targets/Xamarin.Shared.Sdk.targets(1559,3): error :   "_curl_easy_setopt", referenced from: [/Users/user/Desktop/Sentry_test/Sentry_test.csproj]
/usr/local/share/dotnet/packs/Microsoft.macOS.Sdk/14.2.8004/targets/Xamarin.Shared.Sdk.targets(1559,3): error :       _sentry__curl_send_task in libsentry-native.a(sentry_transport_curl.c.o) [/Users/user/Desktop/Sentry_test/Sentry_test.csproj]
/usr/local/share/dotnet/packs/Microsoft.macOS.Sdk/14.2.8004/targets/Xamarin.Shared.Sdk.targets(1559,3): error :   "_curl_easy_strerror", referenced from: [/Users/user/Desktop/Sentry_test/Sentry_test.csproj]
/usr/local/share/dotnet/packs/Microsoft.macOS.Sdk/14.2.8004/targets/Xamarin.Shared.Sdk.targets(1559,3): error :       _sentry__curl_send_task in libsentry-native.a(sentry_transport_curl.c.o) [/Users/user/Desktop/Sentry_test/Sentry_test.csproj]
/usr/local/share/dotnet/packs/Microsoft.macOS.Sdk/14.2.8004/targets/Xamarin.Shared.Sdk.targets(1559,3): error :   "_curl_global_cleanup", referenced from: [/Users/user/Desktop/Sentry_test/Sentry_test.csproj]
/usr/local/share/dotnet/packs/Microsoft.macOS.Sdk/14.2.8004/targets/Xamarin.Shared.Sdk.targets(1559,3): error :       _sentry__curl_bgworker_state_free in libsentry-native.a(sentry_transport_curl.c.o [/Users/user/Desktop/Sentry_test/Sentry_test.csproj]

```

### PR DESCRIPTION
**Pull request by vaind**
- add macos sample - no actual UI at the moment, just inits Sentry. Leaving the UI with some error-capture buttons as a followup for anyone interested in creating one based on https://learn.microsoft.com/en-us/xamarin/mac/get-started/hello-mac
- fix #3070 (see https://github.com/dotnet/runtime/issues/97414 for details)

### CODE REVIEW

File: samples/Sentry.Samples.MacOS/ViewController.cs
Comment (bitsandfoxes) [inline samples/Sentry.Samples.MacOS/ViewController.cs:1]: Why does this one have tabs and the others are spaces..?

File: samples/Sentry.Samples.MacOS/ViewController.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+	public override NSObject RepresentedObject {
+		get => base.RepresentedObject;
+		set {
+			base.RepresentedObject = value;
+
+			// Update the view, if already loaded.
+		}
+	}
+}
```
Comment (bitsandfoxes) [inline samples/Sentry.Samples.MacOS/ViewController.cs:27]:
```suggestion
public partial class ViewController : NSViewController
{
    protected ViewController (NativeHandle handle) : base (handle)
    {
        // This constructor is required if the view controller is loaded from a xib or a storyboard.
        // Do not put any initialization here, use ViewDidLoad instead.
    }

    public override void ViewDidLoad ()
    {
        base.ViewDidLoad ();

        // Do any additional setup after loading the view.
    }

    public override NSObject RepresentedObject {
        get => base.RepresentedObject;
        set {
            base.RepresentedObject = value;

            // Update the view, if already loaded.
        }
    }
}
```

File: samples/Sentry.Samples.macOS/ViewController.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+	public override NSObject RepresentedObject {
+		get => base.RepresentedObject;
+		set {
+			base.RepresentedObject = value;
+
+			// Update the view, if already loaded.
+		}
+	}
+}
```
Comment (bitsandfoxes) [inline samples/Sentry.Samples.macOS/ViewController.cs:27]:
```suggestion
using ObjCRuntime;

namespace Sentry.Samples.MacOS;

public partial class ViewController : NSViewController
{
    protected ViewController (NativeHandle handle) : base (handle)
    {
        // This constructor is required if the view controller is loaded from a xib or a storyboard.
        // Do not put any initialization here, use ViewDidLoad instead.
    }

    public override void ViewDidLoad ()
    {
        base.ViewDidLoad ();

        // Do any additional setup after loading the view.
    }

    public override NSObject RepresentedObject {
        get => base.RepresentedObject;
        set {
            base.RepresentedObject = value;

            // Update the view, if already loaded.
        }
    }
}
```

File: src/Sentry/Platforms/Native/buildTransitive/Sentry.Native.targets [hunk 3]
```diff
@@ -17,12 +17,12 @@
   <ItemGroup Condition="$(TargetFramework.StartsWith('net8')) and '$(OutputType)' == 'Exe' And '$(RuntimeIdentifier)' == 'linux-x64'">
     <DirectPInvoke Include="sentry-native" />
     <NativeLibrary Include="$(MSBuildThisFileDirectory)..\sentry-native\linux-x64\libsentry-native.a" />
-    <LinkerArg Include="-lcurl"/>
+    <NativeSystemLibrary Include="curl" />
```
Comment (bitsandfoxes) [inline src/Sentry/Platforms/Native/buildTransitive/Sentry.Native.targets:20]: Ha. That's the gotcha?

**Review by bitsandfoxes — APPROVED** [review]
Thanks!

File: src/Sentry/Platforms/Native/buildTransitive/Sentry.Native.targets (hunk 3 above)
Comment (vaind) [reply src/Sentry/Platforms/Native/buildTransitive/Sentry.Native.targets:20]: see https://github.com/dotnet/runtime/issues/97414 for details, just added it to the description too

File: samples/Sentry.Samples.MacOS/ViewController.cs
Comment (bruno-garcia) [reply samples/Sentry.Samples.MacOS/ViewController.cs:1]: Xamarin/Mono folks use tabs, ideally we re-write everything

File: src/Sentry/Platforms/Native/buildTransitive/Sentry.Native.targets (hunk 3 above)
Comment (bruno-garcia) [inline src/Sentry/Platforms/Native/buildTransitive/Sentry.Native.targets:20]:
```suggestion
    <!-- See: https://github.com/dotnet/runtime/issues/97414 -->
    <NativeSystemLibrary Include="curl" />
```

**Review by bruno-garcia — APPROVED** [review]
nit: please replace tabs with spaces to align with the rest of the code base

**Review by jamescrosswell — APPROVED** [review]

**Review by bitsandfoxes — APPROVED** [review]

**Pull request merged**